lib/hammer_cli_katello/search_options_creators.rb

```diff
@@ -4,6 +4,13 @@ module HammerCLIKatello
   module SearchOptionsCreators
     include HammerCLIKatello::ForemanSearchOptionsCreators
 
+    def create_repository_sets_search_options(options, mode = nil)
+      create_search_options(options, api.resource(:repository_sets), mode).merge(
+        'product_id' => options[HammerCLI.option_accessor_name('product_id')],
+        'organization_id' => options[HammerCLI.option_accessor_name('organization_id')]
+      )
```

Comment:
This is the change that actually fixes the bug. The validate_options above is for giving nice error messages to the user if they use the wrong combination of options.